cpp/models/lct_secir/model.h

```diff
@@ -35,233 +35,372 @@ namespace mio
+                              Eigen::Ref<Eigen::VectorXd> dydt) const
+    {
+        using LctState = typename std::tuple_element_t<Group, tupleLctStates>;
+
+        size_t first_index = this->populations.template get_first_index_group<Group>();
+        auto params        = this->parameters;
+        ScalarType flow    = 0;
+
+        // Indizes of first subcompartment of the InfectionState for the group in the vectors.
+        size_t Ei_first_index = first_index + LctState::template get_first_index<InfectionState::Exposed>();
+        size_t INSi_first_index =
+            first_index + LctState::template get_first_index<InfectionState::InfectedNoSymptoms>();
+        size_t ISyi_first_index  = first_index + LctState::template get_first_index<InfectionState::InfectedSymptoms>();
+        size_t ISevi_first_index = first_index + LctState::template get_first_index<InfectionState::InfectedSevere>();
+        size_t ICri_first_index  = first_index + LctState::template get_first_index<InfectionState::InfectedCritical>();
+        size_t Ri                = first_index + LctState::template get_first_index<InfectionState::Recovered>();
+        size_t Di                = first_index + LctState::template get_first_index<InfectionState::Dead>();
 
-        // E'
-        dydt[1] = -dydt[0];
-        for (size_t i = 0; i < LctState::template get_num_subcompartments<InfectionState::Exposed>(); i++) {
-            // Dummy stores the value of the flow from dydt[1 + i] to dydt[2 + i].
-            // 1+i is always the index of a (sub-)compartment of E and 2+i can also be the index of the first (sub-)compartment of C.
+        // Calculate derivative of the Susceptible compartment.
+        interact<Group, 0>(pop, y, t, dydt);
+
+        // Calculate derivative of the Exposed compartment.
+        dydt[Ei_first_index] = -dydt[first_index];
+        for (size_t subcomp = 0; subcomp < LctState::template get_num_subcompartments<InfectionState::Exposed>();
+             subcomp++) {
+            // Variable flow stores the value of the flow from one subcompartment to the next one.
+            // Ei_first_index+subcomp is always the index of a (sub-)compartment of Exposed and Ei_first_index
+            // + subcomp + 1 can also be the index of the first (sub-)compartment of InfectedNoSymptoms.
             flow = (ScalarType)LctState::template get_num_subcompartments<InfectionState::Exposed>() *
-                   (1 / params.template get<TimeExposed>()) * y[1 + i];
-            // Subtract flow from dydt[1 + i] and add to dydt[2 + i].
-            dydt[1 + i] = dydt[1 + i] - flow;
-            dydt[2 + i] = flow;
+                   (1 / params.template get<TimeExposed>()[Group]) * y[Ei_first_index + subcomp];
+            // Subtract flow from dydt[Ei_first_index + subcomp] and add to next subcompartment.
+            dydt[Ei_first_index + subcomp]     = dydt[Ei_first_index + subcomp] - flow;
```

Comment:
```suggestion
            dydt[Ei_first_index + subcomp]     -= flow;
```
This is what you did for the other infection states.